### PR TITLE
[FEM.elastic] Some optimisation to make FastTetrahedralCorotational even faster

### PR DESCRIPTION
**Pull request by epernod**
Some small optimisation, thanks for all advices using VS Profiler and SofaBenchmarks

SofaBenchmark have been added on TetraXX FEMForcefield in https://github.com/alxbilger/SofaBenchmark/pull/23

```
-----------------------------------------------------------------------------------------------------------------------------
Benchmark                                                                   Time             CPU   Iterations UserCounters...
-----------------------------------------------------------------------------------------------------------------------------
BM_Scene_bench_StepFactor<TetrahedralFEMForceFieldOptimScene>/512        1339 ms         1344 ms            1 FPS=381.023/s frame=2.62451ms
BM_Scene_bench_StepFactor<TetrahedralFEMForceFieldOptimScene>/1024       2662 ms         2656 ms            1 FPS=385.506/s frame=2.59399ms
BM_Scene_bench_StepFactor<TetrahedralFEMForceFieldOptimScene>/2048       5259 ms         5266 ms            1 FPS=388.938/s frame=2.57111ms
BM_Scene_bench_StepFactor<TetrahedralFEMForceFieldOptimScene>/4096      10451 ms        10453 ms            1 FPS=391.845/s frame=2.55203ms
```

```
-----------------------------------------------------------------------------------------------------------------------------
Benchmark                                                                   Time             CPU   Iterations UserCounters...
-----------------------------------------------------------------------------------------------------------------------------
BM_Scene_bench_StepFactor<TetrahedralFEMForceFieldOptimScene>/512         982 ms          984 ms            1 FPS=520.127/s frame=1.92261ms
BM_Scene_bench_StepFactor<TetrahedralFEMForceFieldOptimScene>/1024       1966 ms         1953 ms            1 FPS=524.288/s frame=1.90735ms
BM_Scene_bench_StepFactor<TetrahedralFEMForceFieldOptimScene>/2048       3988 ms         3969 ms            1 FPS=516.031/s frame=1.93787ms
BM_Scene_bench_StepFactor<TetrahedralFEMForceFieldOptimScene>/4096       7858 ms         7859 ms            1 FPS=521.161/s frame=1.91879ms
```

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/FastTetrahedralCorotationalForceField.inl [hunk 1]
```diff
@@ -458,14 +463,15 @@ void FastTetrahedralCorotationalForceField<DataTypes>::addDForce(const sofa::cor
     const VecMat3x3& edgeDfDx = edgeInfo.getValue();
     Coord deltax;
 
+    auto edges = m_topology->getEdges();
```
Comment (alxbilger) [inline Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/FastTetrahedralCorotationalForceField.inl]:
```suggestion
    const auto& edges = m_topology->getEdges();
```
I prefer to explicit the const-ref

File: Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/FastTetrahedralCorotationalForceField.inl [hunk 2]
```diff
@@ -315,17 +315,22 @@ void FastTetrahedralCorotationalForceField<DataTypes>::addForce(const sofa::core
 
 
     Coord displ[6],sv;
-    Mat3x3 deformationGradient,S,R;
+    Mat3x3NoInit deformationGradient,S,R;
+    auto tetrahedra = m_topology->getTetrahedra();
```
Comment (alxbilger) [inline Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/FastTetrahedralCorotationalForceField.inl]:
```suggestion
    const auto& tetrahedra = m_topology->getTetrahedra();
```

**Review by alxbilger — APPROVED** [review]
Yeeeeeessss!

(minor suggestion then approved)

**Comment by epernod** [comment]
[ci-build][with-all-tests]

**Comment by alxbilger** [comment]
If no objection, I think we can fast merge it @fredroy @hugtalbot

**Comment by fredroy** [comment]
The scene (with this FEM) is OK, so good to go for me

**Pull request merged**